razorpay/razorpay.php

```diff
@@ -123,8 +122,29 @@ public function execPayment($cart)
         $checkoutUrl = 'https://checkout.razorpay.com/v1/checkout.js';
         $amount = number_format($cart->getOrderTotal(true, 3), 2, '.', '')*100;
 
+        // Create order using Orders API right here
+        $keyId = Configuration::get('RAZORPAY_KEY_ID');
+        $keySecret = Configuration::get('RAZORPAY_KEY_SECRET');
+
+        $api = new \Razorpay\Api\Api($keyId, $keySecret);
+
+        $data = $this->getOrderCreationData($cart->id, $amount, $order_currency);
+
+        try
+        {
+            $razorpay_order = $api->order->create($data);
+        }
+        catch (Exception $e)
```

Comment:
Catch Razorpay\Error and Exception separately. Give the exception message in the first case and a generic error in the second. First is our fault (and we should handle it) and the second is beyond our control.